### PR TITLE
Remove `new_qenv`

### LINKED ISSUE #215: Review Deprecated Functions
### Summary

Let's review the functions we deprecated in the previous release and decide whether to hard deprecate them (`lifecycle::deprecate_stop`) or remove them from the codebase entirely.

| Function | Action |
| --- | --- |
| [new_qenv](https://github.com/insightsengineering/teal.code/blob/2c580e9c6fe46190be3c6c44047e553ac994721e/R/qenv-constructor.R#L53-L56) | deprecate_stop |

### PR DESCRIPTION
**Pull request by llrs-roche**
# Pull Request

As commented on #215, it was hard to see a pattern on what to do when deprecating functions.
As `new_qenv()` was important, I decided to keep the code/help page and remove it later on. 

I considered where to document the function; I think having a separate help page will be easier if other methods are deprecated.
Removing all the arguments from the deprecated function was an idea I had because they won't be used, but this created problems with generic not matching actual method, so I left it as is. 

I'm having some issues with pre-commit and it didn't pass the checks, but I think this PR is minimal and won't be any issues

<details>
<summary>Error messages running pre-commit</summary>

```
[INFO] Installing environment for https://github.com/lorenzwalthert/precommit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('C:/PROGRA~1/R/R-44~1.1\\bin\\Rscript.exe', '--vanilla', 'C:\\Users\\<user>\\AppData\\Local\\Temp\\tmpgu57p6xf\\script.R')
return code: 1
stdout:
    # Bootstrapping renv 1.0.7 ---------------------------------------------------
    - Downloading renv ... FAILED
stderr:
    Warning: unable to access index for repository https://packagemanager.posit.co/cran/2024-09-12/bin/windows/contrib/4.4:
      cannot open URL 'https://packagemanager.posit.co/cran/2024-09-12/bin/windows/contrib/4.4/PACKAGES'
    Warning: unable to access index for repository https://packagemanager.posit.co/cran/2024-09-12/src/contrib:
      cannot open URL 'https://packagemanager.posit.co/cran/2024-09-12/src/contrib/PACKAGES'
    Error in h(simpleError(msg, call)) : failed to download:
    All download methods failed
    Calls: source ... bootstrap -> withCallingHandlers -> renv_bootstrap_download
    Execution halted
Check the log at C:\Users\<user>\.cache\pre-commit\pre-commit.log
````

</details>

Fixes #215

### CODE REVIEW

**Comment by llrs-roche** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by m7pr** [comment]
@llrs-roche `new_qenv()` was important but no longer is. We want to simplify the package and support only the new API.
Please also remove `new_qenv()` and let's just keep `qenv()` as the constructor.

After the upcoming release we might not have another release for a very long time. We had last release on CRAN in January 2024. I don't want to wait another year to get this function removed. I think 1 year is enough : )

File: R/qenv-constructor.R [hunk 1]
```diff
@@ -10,13 +10,11 @@
 #' `qenv()` instantiates a `qenv` with an empty environment.
 #' Any changes must be made by evaluating code in it with `eval_code` or `within`, thereby ensuring reproducibility.
 #'
-#' `new_qenv()` (`r badge("deprecated")` and not recommended)
-#' can instantiate a `qenv` object with data in the environment and code registered.
-#'
 #' @name qenv
 #'
-#' @return `qenv` and `new_qenv` return a `qenv` object.
+#' @return `qenv` returns a `qenv` object.
```
Comment (m7pr) [inline R/qenv-constructor.R:15]:
```suggestion
#' @return `qenv()` returns a `qenv` object.
```

Or simplify to "@return `qenv` object" as there's gonna be just `qenv` documented on this site

File: R/qenv-constructor.R [hunk 2]
```diff
@@ -10,13 +10,11 @@
...
 #'
+#' @seealso [`base::within()`], [`get_var()`], [`get_env()`], [`get_warnings()`], [`join()`], [`concat()`]
```
Comment (m7pr) [inline R/qenv-constructor.R:17]: I'm not sure about it. Why not `get_code`, `eval_code` : ) ?
And if you want to list them all, what's the point? There is `index` down below at the end of the help page, and we also have a vignette with the full documentation of `qenv` usage. I think in this case this `@seealso` is redundant. Please remove or please convince me it has any benefit.

File: R/qenv-constructor.R (hunk 1 above)
Comment (llrs-roche) [reply R/qenv-constructor.R:15]: Took the second approach

File: R/qenv-constructor.R [hunk 3]
```diff
@@ -29,29 +27,28 @@ qenv <- function() {
...
 #' @export
 setGeneric("new_qenv", function(env = new.env(parent = parent.env(.GlobalEnv)), code = character()) {
-  lifecycle::deprecate_warn(when = " 0.5.0", what = "new_qenv()", with = "qenv()", always = TRUE)
+  lifecycle::deprecate_stop(
+    when = " 0.5.0",
+    what = "new_qenv()",
+    with = "qenv()"
+  )
   standardGeneric("new_qenv")
 })
```
Comment (m7pr) [inline R/qenv-constructor.R:54]: All below methods should be removed if you decided to go with `lifecycle::deprecate_stop(`.
But I would go one step further and even delete the generic

**Review by m7pr — CHANGES_REQUESTED** [review]
Few changes requested

File: R/qenv-constructor.R (hunk 2 above)
Comment (llrs-roche) [reply R/qenv-constructor.R:17]: These are just the ones there were on the original help page I just moved them around. I didn't want to increase the scope of the PR. But I can delete this if you want, but I don't think it hurts to have more links to other functions from the package that help the user to know how to use the class created using that document.

`eval_code()` and `get_code()` are documented on this same help page, so it doesn't make sense to have a link to the same page.

**Comment by m7pr** [comment]
@llrs-roche one of the CI checks is failing. Can you check what's going on :) ? and what changes you need to provide to push this through the finish line?

File: R/qenv-constructor.R (hunk 2 above)
Comment (m7pr) [reply R/qenv-constructor.R:17]: Alrighty. This make sense! I am not sure about `base::within` though

**Comment by llrs-roche** [comment]
Fixed the issue with the failing test on pkgdown. This check is the one I have as a hook on pre-push on my projects 😄 

For finalizing the PR, I should make sure of teal.gallery work well with the new changes (and other code/packages), and probably also check that it meets the Acceptance Criteria.

**Comment by m7pr** [comment]
@llrs-roche yes please check tlg-catalog and teal.gallery so that they do not use `new_qenv`. Thanks

File: R/qenv-constructor.R (hunk 2 above)
Comment (m7pr) [reply R/qenv-constructor.R:17]: What are your thoughts about `base::within` @llrs-roche ?

**Comment by m7pr** [comment]
I think `new_qenv` does not appear in `teal.gallery` nor in `tlg-catalog`
https://github.com/search?q=repo%3Ainsightsengineering%2Fteal.gallery%20new_qenv&type=code
https://github.com/search?q=repo%3Ainsightsengineering%2Ftlg-catalog++new_qenv&type=code

**Review by m7pr — APPROVED** [review]
Hey @llrs-roche - last comment in here https://github.com/insightsengineering/teal.code/pull/222#discussion_r1827649564

Otherwise this is good to go

File: R/qenv-constructor.R (hunk 2 above)
Comment (m7pr) [reply R/qenv-constructor.R:17]: alrighty

**Pull request merged**